spec/datadog/kit/events_spec.rb

```diff
@@ -10,72 +10,95 @@
 
-  describe '#track_login' do
-    context 'login success' do
-      it 'sets event tracking key on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('appsec.events.users.login.success.track' => 'true')
+  describe '#track_login_success' do
+    it 'sets event tracking key on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' })
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('appsec.events.users.login.success.track' => 'true')
+    end
 
-      it 'sets user id on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('usr.id' => '42')
+    it 'sets user id on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' })
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('usr.id' => '42')
+    end
 
-      it 'sets other keys on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: true, foo: 'bar')
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('usr.id' => '42', 'appsec.events.users.login.success.foo' => 'bar')
+    it 'sets other keys on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_success(trace_op, user: { id: '42' }, foo: 'bar')
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('usr.id' => '42', 'appsec.events.users.login.success.foo' => 'bar')
     end
+  end
 
-    context 'login failure' do
-      it 'sets event tracking key on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: false)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('appsec.events.users.login.failure.track' => 'true')
+  describe '#track_login_failure' do
+    it 'sets event tracking key on trace' do
+      trace_op.measure('root') do
+        described_class.track_login_failure(trace_op, user_id: '42', user_exists: true)
       end
+      trace = trace_op.flush!
+      expect(trace.send(:meta)).to include('appsec.events.users.login.failure.track' => 'true')
+    end
 
-      it 'sets user id on trace' do
-        trace_op.measure('root') do
-          described_class.track_login(trace_op, user: { id: '42' }, success: false)
-        end
-        trace = trace_op.flush!
-        expect(trace.send(:meta)).to include('appsec.events.users.login.failure.usr.id' => '42')
+    it 'sets user id on trace' do
```

Comment:
Minor: The test description here is exactly the same as in track_login_success ("sets user id on trace") and yet we set different tags on both. I would suggest maybe changing the test description on this one ("sets the appsec failure user id on the trace" or something similar) to highlight this fact :)